src/extension_instance.mli

```diff
@@ -0,0 +1,20 @@
+open Import
+
+type t =
```

Comment:
Is it possible to make this type abstract?  I'd prefer if mutability wasn't part of the public api.